### DISCUSSION TITLE
calculate PMF from Umbrella sampling.

### DISCUSSION THREAD

**Discussion by Adityapanda1992** [Colvars/colvars#725] [Q&A] [answered]
Dear developer,
I wonder if Colvar Gomacs's implementation has any way to calculate PMF from umbrella sampling. my colvar file is given below. any help is appreciated greatly.
```
colvar {
    name myColvar
    distanceZ {
        main { atomNumbers 6991 6992 6993 6994 6995 6996 6997 6998 6999 7000 7001 7002 7003 7004 7005 7006 7007}
        ref { dummyAtom (0.32, 0.34, 0.47)}
		oneSiteTotalForce yes
		forceNoPBC yes
    }
}

harmonic {
  colvars myColvar
  forceConstant 20
  centers 3.5         # initial distance
  targetCenters 10.0  # final distance
  targetNumStages 30
  outputCenters yes		# position of restrain center written to trajectory
  targetNumSteps 50000
  outputEnergy yes
```

**Comment by giacomofiorin** [comment, marked as answer]
Hi @Adityapanda1992, first a disclaimer: this simulation is way too short (50000 steps per window) to give you any statistically meaningful results.  Therefore, I assume you are only inquiring about the general protocol and not for this specific run.

What you will need is the `.colvars.traj` file generated during this run (or the set of them if continuing from checkpoints), to be run through an unbiasing method such as WHAM or other methods.  The specifics of the input format depend on each code.  The `.colvars.traj` file will contain the minimum common denominator, i.e. the biased variable and (given that you have `outputEnergy yes`) the corresponding restraint energy.

To extract the trajectories of the `i`-th individual window using the script that you find in `colvartools`, you could do the following:
```bash
python3 plot_colvars_traj.py \
    --first $((${targetNumSteps}*${i})) \
    --last $((${targetNumSteps}*(${i}+1))) \
    --variables myColvar \
    --output-file window-${i}.dat \
    *.colvars.traj
```
or also [load the script in Python](https://colvars.github.io/gromacs-2024/colvars-refman-gromacs.html#sec:colvars_traj_format) and do the slicing there.

@fabsugar You want to provide some example for the FCAM unbiasing method?

**Reply by Adityapanda1992** [reply]
Thank you, @giacomofiorin, for your help. Yes, this is just a demo COLVAR input file as I am trying to learn how to perform Umbrella Sampling (US) with COLVAR. Additionally, I would like to know if COLVAR natively supports any methods for calculating the Potential of Mean Force (PMF). I'm still very new to COLVAR, so I appreciate your assistance once again.

**Reply by giacomofiorin** [reply]
Hi @Adityapanda1992! Among the most commonly used ones in Colvars are ABF ([extended-system version](https://colvars.github.io/gromacs-2024/colvars-refman-gromacs.html#sec:eABF)), [metadynamics](https://colvars.github.io/gromacs-2024/colvars-refman-gromacs.html#sec:colvarbias_meta) and the variations of each.

Additionally, OPES is also upcoming but not merged into a stable release of Colvars or GROMACS yet.

**Reply by Adityapanda1992** [reply]
Thank you for your clarification.
